Redesign devices page

Rishith25 opened the pull request:
## Proposed Changes

- Fixes #11401 
- Redesigned the devices page with location tree node for easier searching for users
- @ohcnetwork/care-fe-code-reviewers

Screenshots

![image](https://github.com/user-attachments/assets/8cbc2731-70df-45c7-87c3-bc8537ecb3cb)



## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced location search enabling users to easily clear their selection.
  - Improved device filtering with dynamic location selection, breadcrumb navigation, and an intuitive sidebar.
  - Added a new localized string for the “View Device” label.

- **Style**
  - Updated the layout and styling of search elements and device display cards for a more modern, user-friendly experience.
  - Redesigned device cards with refined spacing, updated status indicators, and a dedicated action button for viewing details.

- **Bug Fixes**
  - Adjusted logic for handling parent organization and location chains, improving data retrieval accuracy.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Rishith25 committed:
feat: Enhance location selection with search functionality

Rishith25 commented:
@rithviknishad @Jacobjeevan Needed backend support [care#2872](https://github.com/ohcnetwork/care/issues/2872)

Rishith25 committed:
update queryParams

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
fix: Update location search message and rename location parameter

Rishith25 commented:
@rithviknishad @Jacobjeevan Backend PR got merged. Now this PR is Ready for review

rithviknishad commented on src/components/Location/LocationSearch.tsx:
```diff
@@ -55,6 +57,17 @@ export function LocationSearch({
+            >
+              <CareIcon icon="l-times" className="h-4 w-4" />
+            </button>
+          )}
```
shadcn button!

rithviknishad commented on src/pages/Facility/settings/devices/DevicesList.tsx:
```diff
@@ -23,33 +26,65 @@ interface Props {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [selectedLocation, setSelectedLocation] = useState<LocationList | null>(
+    null,
+  );
```
switch to using useFilters, this does not reset page to page 1 when searching or applying the location filter

rithviknishad commented on src/pages/Facility/settings/devices/DevicesList.tsx:
```diff
@@ -23,33 +26,65 @@ interface Props {
       queryParams: {
         offset: (page - 1) * limit,
         limit,
+        registered_name: searchQuery || undefined,
+        current_location: selectedLocation?.id || undefined,
```
let's support searching for user friendly name as well, we could have `search_text` as the query param where backend will search against multiple fields

Rishith25 replied:
@rithviknishad should i use searchbymultiplefileds component in FE for registered_name and user_friendly_name or a single search input?

rithviknishad replied:
single search input;

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
refactor: replace button with Button component in LocationSearch and update DevicesList to use useFilters for pagination and search

Rishith25 committed:
simplify queryKey in DevicesList component

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 commented:
@rithviknishad @Jacobjeevan  Backend PR got merged can review this PR

rithviknishad commented on src/components/Location/LocationSearch.tsx:
```diff
@@ -55,6 +59,17 @@ export function LocationSearch({
           aria-expanded={open}
         >
           {stringifyNestedObject(value || { name: "" }) || "Select location..."}
+          {value && (
+            <Button
+              onClick={(e) => {
+                e.stopPropagation();
+                onSelect(null);
+              }}
+              variant="ghost"
+            >
+              <CareIcon icon="l-times" className="h-4 w-4" />
+            </Button>
+          )}
```
the PR title nor body doesn't convey adding clearing location. this is hard for @nihal467 to test things if not mentioned!

Rishith25 replied:
Sorry my bad. Now I have updated the title and description

Jacobjeevan commented on src/components/Location/LocationSearch.tsx:
```diff
@@ -55,6 +59,17 @@ export function LocationSearch({
         >
           {stringifyNestedObject(value || { name: "" }) || "Select location..."}
+          {value && (
+            <Button
```
<img width="290" alt="image" src="https://github.com/user-attachments/assets/272afa9f-595f-456d-a72e-730ac4671709" />

Adjust styling, esp on hover (image).

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Merge branch 'search-in-devices' of https://github.com/Rishith25/care_fe into search-in-devices

Rishith25 committed:
Add hover effect to LocationSearch button

Jacobjeevan commented on src/pages/Facility/settings/devices/DevicesList.tsx:
```diff
@@ -7,49 +7,83 @@ import CareIcon from "@/CAREUI/icons/CareIcon";
 export default function DevicesList({ facilityId }: Props) {
   const { t } = useTranslation();
-  const [page, setPage] = useState(1);
+  const [selectedLocation, setSelectedLocation] = useState<LocationList | null>(
```
Let's use qParams instead of maintaining a separate state for location.

Plus queryKey needed to include location anyways...but above will take care of that.

Rishith25 replied:
@Jacobjeevan if i use qParams for selectedLocation, then we cant pass location value to `LocationSearch ` componenet as it takes location value not id.

```.tsx  
<LocationSearch
              facilityId={facilityId}
              onSelect={setSelectedLocation}
              value={selectedLocation}
            />
```

We can use qParams for location but we need to still use selectedLocation State to pass it to `LocationSearch ` component

Once see below code

```.tsx
<LocationSearch
              facilityId={facilityId}
              onSelect={(location: LocationList | null) => {
                updateQuery({
                  current_location: location?.id || undefined,
                });
                setSelectedLocation(location);
              }}
              value={selectedLocation}
            />
```
          
Which approach u prefer?

Jacobjeevan replied:
Ok, that's fine then. Latest changes are good.

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Update current_location query parameter in DevicesList

Jacobjeevan reviewed (changes requested):
Let's adjust the style. Right now the select looks like it's disabled (adjust the bg, maybe add a border so it looks similar to the search input).

<img width="573" alt="image" src="https://github.com/user-attachments/assets/12cdcc02-b8d0-4699-8629-dff7daa214ea" />

Also adjust the trigger width in location search (you can look into shadcn's documentation and other examples of popover in the codebase on how to do this). Make sure to test the other components where LocationSearch is called.

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
style: Improve LocationSearch component styling and placeholder text

nihal467 commented:
![image](https://github.com/user-attachments/assets/4ee9da42-28d2-4639-8b58-e0db8a89ecb7)
![image](https://github.com/user-attachments/assets/aa0f21d6-797c-4385-81a9-1eea8e141dcd)

- Keeping the name as a searchable text box and the location as a dropdown selector is not consistent with our platform's design. We should combine them into one search area
- Align it with the encounter page search, where users type to filter data and select name or location from badges within the search component.

@Rishith25 when working on elements commonly used across the platform, ensure UI consistency

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/420876669-4ee9da42-28d2-4639-8b58-e0db8a89ecb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MDc4OTQsIm5iZiI6MTc0MTYwNzU5NCwicGF0aCI6Ii81NzA1NTk5OC80MjA4NzY2NjktNGVlOWRhNDItMjhkMi00NjM5LThiNTgtZTBkYjhhODllY2I3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDExNTMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhZGIwMDY4Mjg5YjgxODE0MjhiNTliZDEzNDk3MDIzZGEzZjVkNjg3ZDU2ZjA0NmI3ODY1OTZlNTE5NWM1NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yYUv5zwGAkZKoit0Fcm1gAMri_emlUTYeFXCfI_B5QU) ![image](https://private-user-images.githubusercontent.com/57055998/420876848-aa0f21d6-797c-4385-81a9-1eea8e141dcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MDc4OTQsIm5iZiI6MTc0MTYwNzU5NCwicGF0aCI6Ii81NzA1NTk5OC80MjA4NzY4NDgtYWEwZjIxZDYtNzk3Yy00Mzg1LTgxYTktMWVlYThlMTQxZGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDExNTMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxNDgzYjhmNDVmMTU4N2NlMTc5NGVlMmI5YjkzZTkyNDk0MWM3OTNkOWZlMzUzMGMxMDEyOTJkZTdhNmUxY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8gKFk1RnG7Y8rj4nBimxQ8oZj1lIIxviB0_7Lwpc4zE)
> 
> * Keeping the name as a searchable text box and the location as a dropdown selector is not consistent with our platform's design. We should combine them into one search area
> * Align it with the encounter page search, where users type to filter data and select name or location from badges within the search component.
> 
> @Rishith25 when working on elements commonly used across the platform, ensure UI consistency
> 
> can you share the ETD to get this done

Hey @nihal467 
On using SearchByMultipleField Component we cant show location list to select a location. If we use SearchByMultipleField component then we have to add a location_id as field becz we have to pass location_id as queryparam in api call.

Should i add location_id as search filed ??

cc: @rithviknishad @Jacobjeevan

rithviknishad commented:
@Rishith25 could you rethink the design? 🤔 i don't think this looks good enough to be merged;

Rishith25 commented:
@rithviknishad This is the design given by @vivekrajuv18. 
Could you please confirm to proceed with or not 

![image](https://github.com/user-attachments/assets/46494806-7166-4ffc-bd0f-39070467153d)

![image](https://github.com/user-attachments/assets/a0d56c5d-9275-4c73-a09b-414c7c2c9ab4)

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
feat: Add search type selection for devices list and improve location search component

Rishith25 committed:
refactor: Enhance layout and styling for device search and location components

Rishith25 committed:
refactor: Remove unused translation hook from LocationSearch component

Jacobjeevan commented on src/components/Location/LocationSearch.tsx:
```diff
@@ -50,14 +57,31 @@ export function LocationSearch({
         </div>
       </PopoverTrigger>
-      <PopoverContent className="w-[400px] p-0">
+      <PopoverContent className="min-w-[250px] max-w-[400px] w-full p-0">
```
check `--radix-popover-trigger-width` in Shadcn popover documentation. You can also check codebase for other examples.

Jacobjeevan reviewed (changes requested):
Need to adjust for mobile:

<img width="419" alt="image" src="https://github.com/user-attachments/assets/17e52a64-da4e-4b52-af61-2bd2705b5611" />

Move to next row as it looks for smaller devices.

<img width="546" alt="image" src="https://github.com/user-attachments/assets/09a1f467-d532-4345-b473-17646188d543" />

Also adjust the size, so it matches the container for tablets:

<img width="574" alt="image" src="https://github.com/user-attachments/assets/7d0c205a-c972-4112-bc75-ac6b9fc5165c" />

Rishith25 committed:
Merge branch 'develop' of github.com:ohcnetwork/care_fe into search-in-devices

Rishith25 committed:
Refactor DevicesList component layout and integrate Page wrapper

Rishith25 committed:
Update PopoverContent width in LocationSearch component

Jacobjeevan commented on src/pages/Facility/settings/devices/DevicesList.tsx:
```diff
@@ -7,82 +7,154 @@ import CareIcon from "@/CAREUI/icons/CareIcon";
+                  value={searchType}
+                  onValueChange={(value: "name" | "location") => {
+                    setSearchType(value);
+                    updateQuery({
```
This results in slightly weird behavior when we switch back and forth.

Searching a name and then switching to location would preserve the name (for this, we will need to add badges to ensure it's clear to user that search is preserved).

However if user were to select a location and then switch back to name again, name would reset.

Vice versa for location as well.

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 commented:
@rithviknishad i have done the things you mentioned in https://github.com/ohcnetwork/care_fe/issues/11401#issuecomment-2745321698 in this PR

CC : @Jacobjeevan

Rishith25 committed:
Merge branch 'develop' into search-in-devices

rithviknishad commented:
> @rithviknishad i have done the things you mentioned in [#11401 (comment)](https://github.com/ohcnetwork/care_fe/issues/11401#issuecomment-2745321698) in this PR
> 
> CC : @Jacobjeevan

you can do the #11401 (redesign) issue itself in this PR as well, the search and care type filters are anyways already there in develop right now.

rithviknishad commented:
also, we'll need to be able to filter both by location and name together as well!

Rishith25 commented:
> also, we'll need to be able to filter both by location and name together as well!

Yes I have done that in this PR. Plz take a look once

Rishith25 commented:
![Screenshot_20250324_093451.jpg](https://github.com/user-attachments/assets/5933a2b4-04f3-4dde-b2ca-73a69cfa17e0)

@rithviknishad

rithviknishad commented:
I don't think this is a good UX, users may not be aware that it can be filtered by both at once. They might see this as either Name or Location filter.

Also can we hide the filter by type filter when there are no types registered by plugs?

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Add "View Device" text and enhance DeviceCard layout for better usability

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Refactor: Improve Popover component structure in DevicesList

rithviknishad commented:
Closing as no updates;

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Refactor DevicesList component: integrate location tree

Rishith25 commented:
@rithviknishad @jacobjeevan
Can you reopen this PR, I have updated pr with latest design
![image](https://github.com/user-attachments/assets/e126eb1b-13cd-4906-8bee-8fb0244fc9e8)

Rishith25 committed:
Merge branch 'develop' into search-in-devices

rithviknishad commented:
is this ready for review? 🤔 do a self-review!

Rishith25 commented:
@rithviknishad This PR is ready for review. I have updated this PR with the new design 

![image](https://github.com/user-attachments/assets/86609ddb-23c1-4877-9d45-c0dc3d951686)


- [x] Added the location filter by which the user can now know that the devices can be filtered with both location as well as name.
- [x] If Plugin devices types length is greater than 0, then the filter by type is visible.

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 committed:
Remove console.log for children in OrganizationTreeNode component

Rishith25 commented:
> Please follow the merge checklist! i18n is checked; but there are missing translations! Also the code has unnecessary console.log statements; self-review not done!
> 
> This has been repeated many times;

@rithviknishad 
Self-review: 
1. I have added i18n translations.
2. Removed the unnecessary console logs and 
3.  Added the location filter by which the user can now know that the devices can be filtered with both location as well as name.
4.  If Plugin devices types length is greater than 0, then the filter by type is visible.

rithviknishad committed:
make parent chain code reusable utility

rithviknishad committed:
Merge branch 'search-in-devices' of github.com:Rishith25/care_fe into search-in-devices

Rishith25 committed:
Merge branch 'develop' into search-in-devices

nihal467 commented:
@Rishith25 can you clear the merge conflict, ASAP

Rishith25 committed:
Merge branch 'develop' into search-in-devices

Rishith25 commented:
> @Rishith25 can you clear the merge conflict, ASAP

@nihal467 Resolved the merge conflicts

nihal467 committed:
Merge branch 'develop' into search-in-devices

nihal467 commented:
https://rebuildearth.slack.com/archives/C080225HC6B/p1741761891632839

- the design and the PR preview is different, follow up the discussion in the slack
- fix the cypress failure

rithviknishad commented:
closing due to no activity and way past the ETD mentioned in the issue
